swift/Sources/SignalClient/DeviceTransfer.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Copyright 2020 Signal Messenger, LLC
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+import SignalFfi
+import Foundation
+
+public func generateDeviceTransferPrivateKey() -> [UInt8] {
+    return failOnError {
+        try invokeFnReturningArray {
+            signal_device_transfer_generate_private_key($0, $1)
+        }
+    }
+}
+
+/*
+public func generateDeviceTransferCertificate<Bytes: ContiguousBytes>(privateKey: Bytes,
+                                                                      name: String,
+                                                                      daysTilExpire: UInt32) -> [UInt32] {
```

Comment:
Return type should be [UInt8]. (That's probably the only problem.)

Reply:
Argument type should be Int too—it's conventional in Swift to use Int everywhere unless you need a specific range, even if a number's required to be non-negative. Converting to UInt32 will do a range check and trap.

Reply:
:joy: Thanks just completely missed this, and the error message wasn't giving any hints.